### PR TITLE
ENG-6607: Introduce support for event cursors

### PR DESCRIPTION
**Pull request by erickpintor**
### Description

Provide support for the new per-event stream cursor.

### Motivation and context

There's an edge case where resuming a stream from a txn time with lots of documents written could result in event loss. This approach fixes that edge case.

### How was the change tested?

Unit tests included.

### Screenshots (if appropriate):

NA

### Change types

* - [X] Bug fix (non-breaking change that fixes an issue)
* - [X] New feature (non-breaking change that adds functionality)
* - [ ] Breaking change (backwards-incompatible fix or feature)

### Checklist:

* - [X] My code follows the code style of this project.
* - [X] My change requires a change to Fauna documentation.
* - [X] My change requires a change to the README, and I have updated it accordingly.

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jrodewig — DISMISSED** [review]
Narrow 👍  on README and comments.

File: fauna/client/client.py [hunk 1]
```diff
@@ -523,6 +531,7 @@ def _next_element(self):
           FaunaError.parse_error_and_throw(event, 400)
 
         self.last_ts = event["txn_ts"]
+        self.last_cursor = event["cursor"]
```
Comment (marrony) [inline fauna/client/client.py:534]: looking the tests failures, looks like `cursor` is not present in all events, you might need to change to `event.get("cursor")`

Comment (erickpintor) [reply fauna/client/client.py:534]: Yeah. That and also I forgot to update the mocked tests somehow.

**Review by marrony — DISMISSED** [review]

**Review by pnwpedro — APPROVED** [review]
